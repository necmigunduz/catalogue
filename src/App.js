import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Catalogue Project</h1>
    </div>
  );
}

export default App;
