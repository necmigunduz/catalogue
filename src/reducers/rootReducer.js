import { combineReducers } from 'redux';
import filterReducer from './filtersReducer';
import mealReducer from './meal';
import mealsReducer from './meals';

const rootReducer = combineReducers(
  {
    filters: filterReducer,
    meal: mealReducer,
    meals: mealsReducer,
  },
);

export default rootReducer;
