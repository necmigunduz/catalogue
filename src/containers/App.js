import { BrowserRouter, Switch, Route } from 'react-router-dom';
// import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
import '../assets/styles/App.css';
import MealsList from './MealsList';
import FoodsInfo from './FoodsInfo';

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route
          exact
          path="/"
          render={() => (
            <MealsList />
          )}
        />
        <Route
          exact
          path="/recipe"
          render={() => (
            <FoodsInfo />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
