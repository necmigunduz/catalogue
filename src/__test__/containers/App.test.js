import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import PropTypes from 'prop-types';
import App from '../../containers/App';
import rootReducer from '../../reducers/rootReducer';

const state = {
    meals: [],
    meal: {},
    filters: {
      q: '',
      dietLabels: '',
      healthLabels: '',
      cuisineType: '',
      dishType: '',
      mealType: '',
    },
  };
  
  const store = createStore(rootReducer, state);
  
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  
  Wrapper.propTypes = {
    children: PropTypes.shape({}).isRequired,
  };
  
  describe('App', () => {
    it('renders properly', () => {
      const tree = render(<App />, { wrapper: Wrapper });
      expect(tree).toMatchSnapshot();
    });
  });
  
