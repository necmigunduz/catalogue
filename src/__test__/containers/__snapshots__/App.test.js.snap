// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <input
            id="q"
            placeholder="Enter ingredients and cooking equipment to search for a recipe including them"
            type="text"
            value=""
          />
          <div>
            <div>
              <select
                id="dietLabels"
              >
                <option>
                  Diet
                </option>
                <option>
                  Balanced
                </option>
                <option>
                  High-fiber
                </option>
                <option>
                  High-protein
                </option>
                <option>
                  Low-carb
                </option>
                <option>
                  Low-fat
                </option>
                <option>
                  Low-sodium
                </option>
              </select>
              <div />
            </div>
            <div>
              <select
                id="healthLabels"
              >
                <option>
                  Health
                </option>
                <option>
                  Alcohol-free
                </option>
                <option>
                  Celery-free
                </option>
                <option>
                  Crustacean-free
                </option>
                <option>
                  Dairy-free
                </option>
                <option>
                  Egg-free
                </option>
                <option>
                  Fish-free
                </option>
                <option>
                  Fodmap-free
                </option>
                <option>
                  Gluten-free
                </option>
                <option>
                  Immuno-supportive
                </option>
                <option>
                  Keto-friendly
                </option>
                <option>
                  Kidney-friendly
                </option>
                <option>
                  Kosher
                </option>
                <option>
                  Low-fat-tabs
                </option>
                <option>
                  Low-potasium
                </option>
                <option>
                  Low-sugar
                </option>
                <option>
                  Lupine-free
                </option>
                <option>
                  Mediterranean
                </option>
                <option>
                  Mustard-free
                </option>
                <option>
                  No-oil-added
                </option>
                <option>
                  Paleo
                </option>
                <option>
                  Peanut-free
                </option>
                <option>
                  Pescatarian
                </option>
                <option>
                  Pork-free
                </option>
                <option>
                  Red-meat-free
                </option>
                <option>
                  Sesame-free
                </option>
                <option>
                  Shellfish-free
                </option>
                <option>
                  Soy-free
                </option>
                <option>
                  Sugar-conscious
                </option>
                <option>
                  Tree-nut-free
                </option>
                <option>
                  Vegan
                </option>
                <option>
                  Vegetarian
                </option>
                <option>
                  Wheat-free
                </option>
              </select>
              <div />
            </div>
            <div>
              <select
                id="cuisineType"
              >
                <option>
                  Cuisine
                </option>
                <option>
                  American
                </option>
                <option>
                  Asian
                </option>
                <option>
                  British
                </option>
                <option>
                  Caribbean
                </option>
                <option>
                  Central Europe
                </option>
                <option>
                  Chinese
                </option>
                <option>
                  Eastern Europe
                </option>
                <option>
                  French
                </option>
                <option>
                  Indian
                </option>
                <option>
                  Italian
                </option>
                <option>
                  Japanese
                </option>
                <option>
                  Kosher
                </option>
                <option>
                  Mediterranean
                </option>
                <option>
                  Mexican
                </option>
                <option>
                  Middle Eastern
                </option>
                <option>
                  Nordic
                </option>
                <option>
                  South American
                </option>
                <option>
                  South East Asian
                </option>
              </select>
              <div />
            </div>
            <div>
              <select
                id="dishType"
              >
                <option>
                  Dish
                </option>
                <option>
                  alcohol
                </option>
                <option>
                  biscuits and cookies
                </option>
                <option>
                  bread
                </option>
                <option>
                  cereals
                </option>
                <option>
                  condiments and sauces
                </option>
                <option>
                  drinks
                </option>
                <option>
                  desserts
                </option>
                <option>
                  egg
                </option>
                <option>
                  main course
                </option>
                <option>
                  omelet
                </option>
                <option>
                  pancake
                </option>
                <option>
                  preps
                </option>
                <option>
                  preserve
                </option>
                <option>
                  salad
                </option>
                <option>
                  sandwiches
                </option>
                <option>
                  soup
                </option>
                <option>
                  starter
                </option>
              </select>
              <div />
            </div>
            <div>
              <select
                id="mealType"
              >
                <option>
                  Meal
                </option>
                <option>
                  Breakfast
                </option>
                <option>
                  Lunch
                </option>
                <option>
                  Dinner
                </option>
                <option>
                  Snack
                </option>
                <option>
                  Teatime
                </option>
              </select>
              <div />
            </div>
          </div>
          <a
            href="#search"
            type="button"
          >
            Click to Search
          </a>
          <button
            id="clear"
            type="button"
          >
            Clear All
          </button>
        </div>
      </div>
      <div>
        <p>
          No recipe found!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <input
          id="q"
          placeholder="Enter ingredients and cooking equipment to search for a recipe including them"
          type="text"
          value=""
        />
        <div>
          <div>
            <select
              id="dietLabels"
            >
              <option>
                Diet
              </option>
              <option>
                Balanced
              </option>
              <option>
                High-fiber
              </option>
              <option>
                High-protein
              </option>
              <option>
                Low-carb
              </option>
              <option>
                Low-fat
              </option>
              <option>
                Low-sodium
              </option>
            </select>
            <div />
          </div>
          <div>
            <select
              id="healthLabels"
            >
              <option>
                Health
              </option>
              <option>
                Alcohol-free
              </option>
              <option>
                Celery-free
              </option>
              <option>
                Crustacean-free
              </option>
              <option>
                Dairy-free
              </option>
              <option>
                Egg-free
              </option>
              <option>
                Fish-free
              </option>
              <option>
                Fodmap-free
              </option>
              <option>
                Gluten-free
              </option>
              <option>
                Immuno-supportive
              </option>
              <option>
                Keto-friendly
              </option>
              <option>
                Kidney-friendly
              </option>
              <option>
                Kosher
              </option>
              <option>
                Low-fat-tabs
              </option>
              <option>
                Low-potasium
              </option>
              <option>
                Low-sugar
              </option>
              <option>
                Lupine-free
              </option>
              <option>
                Mediterranean
              </option>
              <option>
                Mustard-free
              </option>
              <option>
                No-oil-added
              </option>
              <option>
                Paleo
              </option>
              <option>
                Peanut-free
              </option>
              <option>
                Pescatarian
              </option>
              <option>
                Pork-free
              </option>
              <option>
                Red-meat-free
              </option>
              <option>
                Sesame-free
              </option>
              <option>
                Shellfish-free
              </option>
              <option>
                Soy-free
              </option>
              <option>
                Sugar-conscious
              </option>
              <option>
                Tree-nut-free
              </option>
              <option>
                Vegan
              </option>
              <option>
                Vegetarian
              </option>
              <option>
                Wheat-free
              </option>
            </select>
            <div />
          </div>
          <div>
            <select
              id="cuisineType"
            >
              <option>
                Cuisine
              </option>
              <option>
                American
              </option>
              <option>
                Asian
              </option>
              <option>
                British
              </option>
              <option>
                Caribbean
              </option>
              <option>
                Central Europe
              </option>
              <option>
                Chinese
              </option>
              <option>
                Eastern Europe
              </option>
              <option>
                French
              </option>
              <option>
                Indian
              </option>
              <option>
                Italian
              </option>
              <option>
                Japanese
              </option>
              <option>
                Kosher
              </option>
              <option>
                Mediterranean
              </option>
              <option>
                Mexican
              </option>
              <option>
                Middle Eastern
              </option>
              <option>
                Nordic
              </option>
              <option>
                South American
              </option>
              <option>
                South East Asian
              </option>
            </select>
            <div />
          </div>
          <div>
            <select
              id="dishType"
            >
              <option>
                Dish
              </option>
              <option>
                alcohol
              </option>
              <option>
                biscuits and cookies
              </option>
              <option>
                bread
              </option>
              <option>
                cereals
              </option>
              <option>
                condiments and sauces
              </option>
              <option>
                drinks
              </option>
              <option>
                desserts
              </option>
              <option>
                egg
              </option>
              <option>
                main course
              </option>
              <option>
                omelet
              </option>
              <option>
                pancake
              </option>
              <option>
                preps
              </option>
              <option>
                preserve
              </option>
              <option>
                salad
              </option>
              <option>
                sandwiches
              </option>
              <option>
                soup
              </option>
              <option>
                starter
              </option>
            </select>
            <div />
          </div>
          <div>
            <select
              id="mealType"
            >
              <option>
                Meal
              </option>
              <option>
                Breakfast
              </option>
              <option>
                Lunch
              </option>
              <option>
                Dinner
              </option>
              <option>
                Snack
              </option>
              <option>
                Teatime
              </option>
            </select>
            <div />
          </div>
        </div>
        <a
          href="#search"
          type="button"
        >
          Click to Search
        </a>
        <button
          id="clear"
          type="button"
        >
          Clear All
        </button>
      </div>
    </div>
    <div>
      <p>
        No recipe found!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
